skate-plugin/src/main/kotlin/com/slack/sgp/intellij/featureflags/FeatureFlagFileListener.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2023 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.slack.sgp.intellij.featureflags
+
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.fileEditor.FileEditorManagerListener
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.util.PsiUtilBase
+
+/**
+ * Listener that responds to file opening events within the IDE. This listener is interested in
+ * files that match the pattern `.*Features.kt`. Upon detection of such a file opening, it triggers
+ * a PSI analysis and attempts to extract feature flags.
+ */
+class FeatureFlagFileListener : FileEditorManagerListener {
+
+  override fun fileOpened(source: FileEditorManager, file: VirtualFile) {
+    if (isFeatureFile(file)) {
+      val editor = source.selectedTextEditor
+      val psiFile = PsiUtilBase.getPsiFileInEditor(editor!!, source.project)
+      psiFile?.let {
+        val featureFlagHandler = FeatureFlagExtractor()
+        featureFlagHandler.extractFeatureFlags(psiFile)
+      }
+    }
+  }
+
+  private fun isFeatureFile(file: VirtualFile): Boolean {
+    return file.name.matches(Regex(".*Feature\\.kt"))
```

Comment:
Three notes on this
- I don't think we need a regex since this could just be a suffix check?
- let's make this configurable
- for future ref, you generally want to cache Regex instances as it would otherwise recompile the internal regex every time this is called.

Reply:
[0a8fd75](https://github.com/slackhq/slack-gradle-plugin/pull/552/commits/0a8fd75d0b350635dde73d46c101d4048b24c49b)